test/server.test.js

```diff
@@ -2557,3 +2557,57 @@ test('should have proxy event handlers as instance', function(t) {
         t.end();
     });
 });
+
+test('earliest chain should get to reject requests', function(t) {
+    SERVER.get('/foobar', function(req, res, next) {
+        t.fail('should not call handler');
+    });
+
+    SERVER.earliest(function(req, res) {
+        res.statusCode = 413; // I'm a teapot!
+        res.end();
+        return false;
+    });
+
+    CLIENT.get('/foobar', function(_, __, res) {
+        t.equal(res.statusCode, 413);
+        t.end();
+    });
+});
+
+test('earliest chain should get to allow requests', function(t) {
+    SERVER.get('/foobar', function(req, res, next) {
+        res.send(413, 'Im a teapot');
+        return next();
+    });
+
+    SERVER.earliest(function(req, res) {
+        return true;
+    });
+
+    CLIENT.get('/foobar', function(_, __, res) {
+        t.equal(res.statusCode, 413);
+        t.end();
+    });
+});
+
+test('earliest chain should allow multiple handlers', function(t) {
```

Comment:
Looks like we should add a test that confirms inflight numbers are accurate after an earliest rejection event?

Reply:
Ah good point :+1: